rust/lance/src/dataset/take.rs

```diff
@@ -130,6 +129,9 @@ async fn do_take_rows(
     mut builder: TakeBuilder,
     projection: Arc<ProjectionPlan>,
 ) -> Result<RecordBatch> {
+    let projection_with_row_id = projection.physical_projection.with_row_id;
```

Comment:
nit: the variable name is a bit confusing, what about `projection_has_row_id` and `projection_has_row_addr` to be more clear these are booleans

Reply:
What about with_row_id_in_projection